react/components/ProductSummary.js

```diff
@@ -170,8 +187,27 @@ function ProductSummaryWrapper(props) {
   const { isPriceAsync } = useIsPriceAsync()
 
   return (
-    <ProductSummaryProvider {...props} isLoading={isPriceAsync}>
-      <ProductSummaryCustom {...props} />
+    <ProductSummaryProvider {...props} isPriceLoading={isPriceAsync}>
+      <ProductSummaryConsumer>
```

Comment:
Why not use `useProductSummary()`? Isn't it the same? Using a hook would avoid creating another component in the React tree.

Reply:
I refactored the code. Now, this is not necessary anymore. I created the `ProductPriceSimulationWrapper` which is responsible for encapsulating all the hooks, including `useProductSummary()`